### PR TITLE
add `json()` impl to `Request`

### PR DESCRIPTION
**Pull request by minauteur**
In draft until feedback on approach is given in issue #199 thread

### CODE REVIEW

**Comment by hamza1311** [comment]
Sorry, it took me a little bit to get to this. It looks good. You can go ahead and finish it up so it can be merged

**Comment by minauteur** [comment]
No worries!
I'll get this cleaned up and have tests in order so this can be ready for review sometime this evening or tomorrow morning.
Thanks!

**Comment by minauteur** [comment]
I've cleaned up the formatting, and--for now--I've updated the test `post_json()` in `crates/net/tests/http.rs` to utilize the convenience method added in the PR--passes locally with `HTTPBIN_URL` and `ECHO_SERVER_URL` environment variables set accordingly.
Happy to flesh out tests in a later PR if this is sufficient, otherwise, I may need a some guidance on where (an) added test(s) should live for covering this method specifically.

**Comment by minauteur** [comment]
OK--I've added a separate test: `json_body()` to assert that we get a `Result::Ok` back from `Request::json()` when we pass it a serializable type.
I also updated `post_json()` to use the added method for creating its requests.
An additional field: `num: i16` has also been added to the structs used for both tests.
@hamza1311 let me know if you'd like me to add or amend anything--happy to help!

Thanks again for the feedback!f

File: crates/net/src/http/mod.rs [hunk 1]
```diff
@@ -124,7 +125,11 @@ impl Request {
         self.options.integrity(integrity);
         self
     }
-
+    /// convenience method for json request bodies
```
Comment (hamza1311) [inline crates/net/src/http/mod.rs:128]:
```suggestion
    /// A convenience method to set JSON as request body
    ///
    /// # Note
    /// 
    /// This method also sets the `Content-Type` header to `application/json`
```

**Review by hamza1311 — CHANGES_REQUESTED** [review]
Looks good. Thanks for the PR!

Just need to update the documentation and this is good to go

**Comment by minauteur** [comment]
✔️ Thanks so much!

**Comment by hamza1311** [comment]
Thanks for the PR. This is good to go

**Comment by minauteur** [comment]
Awesome--feel free to merge whenever!
Not sure if the comment needs that line deleted from the linting check--happy to do that if so

**Comment by hamza1311** [comment]
Please ensure that the CI is green

**Comment by minauteur** [comment]
> Please ensure that the CI is green

Thanks.
Looks like you (or another maintainer) will need to approve running the workflow, but it should pass with the latest commit!

**Pull request merged**